Userland/Libraries/LibGfx/ImageFormats/CCITTDecoder.cpp

```diff
@@ -436,7 +436,7 @@ ErrorOr<ReferenceLine> decode_single_ccitt_2d_line(BigEndianInputBitStream& inpu
                 reference_line.take_first();
                 continue;
             }
-            if (change.color != current_color)
+            if (change.color != current_color || change.column == image_width)
```

Comment:
Do you know why Tests/LibGfx/test-inputs/tiff/ccitt4.tiff doesn't need this? It has identical colors at the edge.

Reply:
We add a `Change{invert(last_color), m_image_width}` in every `ReferenceLine`. And in ccitt4.tiff, every end of line is encoded with a vertical_0 code. That works well as we are already in white and the last change of the line is a `{Black, 400}`.

However, some images rely on the edge being a change of any color, as an example this image can be encoded as:

L1: WWW
L2: BWB

L1: V_0 # Using the edge as a change to black
L2: V_3, V_2, V_1 # Using the edge alternatively as black or white


The first iteration of this patch was adding a second `Change{invert(last_color), m_image_width}` to the reference line, but it felt a bit more hackish.

Hope it helped you understand!